src/Integration.Vsix/CFamily/CMake/CompilationDatabaseRequest.cs

```diff
@@ -89,11 +99,29 @@ public void WriteRequest(BinaryWriter writer)
 
             // TODO - only supplied if analysing a header file
 //            WriteSetting(writer, "HeaderFilerLanguage", Context.CFamilyLanguage);
-            WriteRuleSettings(writer);
 
             WriteFooter(writer);
         }
 
+        private void WriteRulesConfig(BinaryWriter writer)
+        {
+            // Optimisation - no point in calculating the active rules if we're
+            // creating a pre-compiled header, as they won't be used.
+            // However, the QualityProfile is an essential setting so we still
+            // have to write it.
+            if (!Context.AnalyzerOptions?.CreatePreCompiledHeaders ?? true)
+            {
+                var rulesProtocolFormat = rulesConfigProtocolFormatter.Format(Context.RulesConfiguration);
+
+                WriteQualityProfile(writer, rulesProtocolFormat.QualityProfile);
+                WriteRuleSettings(writer, rulesProtocolFormat.RuleParameters);
+            }
+            else
+            {
+                WriteQualityProfile(writer, "");
```

Comment:
I believe we don't need this, I tested the pch generation and it seems fine without it. Will ask the cfamily.

Reply:
It's listed as an essential field in the CMake integration spec, and their code suggests they will error without it:
```
llvm::Optional<DriverError> MSVCDriver::setCompilationIndependentOpt(AnalyzerInput& analyzerInput) const {
...
  if (auto errorOrOpt = getOptionOrError("QualityProfile")) {
...
}
```

It's possible they are returning an error message and we are ignoring it.